[For Testing] Fix number format propagation

0penBrain opened the pull request.

donovaly commented:
I have problems to apply the PR because if the second commit. Could you please squash then to one commit and rebase?

0penBrain commented:
@donovaly Here it is ;)

donovaly commented:
Hey, this works for me! InputField works as well as App::PropertyFloat.

freecadci commented:
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_6669"><img alt="pipeline status" src="https://gitlab.com/freecad/FreeCAD-CI/badges/PR_6669/pipeline.svg"/></a> for feature branch [PR_6669](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_6669). Pipeline [502424397](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/502424397) was triggered at [6f75227](https://github.com/FreeCAD/FreeCAD/pull/6669/commits/6f752278d533c2000a017997a5c01cba9aa5b708). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

0penBrain commented:
Great. I'll clean up the code on tonight (I just pushed it dirty for testing) and remove draft status. 

@wwmayer for information I also tested with your previous fix enabled in case it would fix the small inconvenience in InputField but it didn't (I also tested another flavor that you still can see as a comment right now). Why it's now disabled. ;)

wwmayer reviewed (commented):
I have tested it on Linux with LANG=de_DE.UTF-8 and LANG=C and everything works as expected.
Additionally, it fixes a minor annoyance: a dialog with InputField doesn't need to be closed and re-opened after a language change. It will automatically use the new locale settings as soon as it contents gets modified.

If on Windows everything still works correctly then I think it's ready for merge.

donovaly commented:
> If on Windows everything still works correctly

From my side yes.

0penBrain commented:
@donovaly @wwmayer I'm ready for this.
I cleaned the existing PR and rebased.
I also made some slight cosmetics on the preferences.
I fixed 2 small bugs (say corner cases) introduced by bba7719cd15e755a5 .
Finally I changed a bit `Translator::SetLocale` so it defaults to OS locale (to be in sync with general behavior) and thus we don't need anymore `Translator::SetSytemLocale`.

If one or more commits are a problem for you to merge, just let me know and I can remove them.

freecadci commented:
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_6669"><img alt="pipeline status" src="https://gitlab.com/freecad/FreeCAD-CI/badges/PR_6669/pipeline.svg"/></a> for feature branch [PR_6669](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_6669). Pipeline [503110212](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/503110212) was triggered at [e26f976](https://github.com/FreeCAD/FreeCAD/pull/6669/commits/e26f9762dfbf133abd25cfe5a537e89019ea80fe). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

wwmayer commented on src/Gui/DlgGeneralImp.cpp:
```diff
@@ -129,19 +129,21 @@ void DlgGeneralImp::setLanguage()
 
     // Only make the change if locale setting has changed
-    if (localeIndex == localeFormat)
+    if (!force && localeIndex == localeFormat)
```
The idea of the class member localeIndex was that if the language has been changed but the locale is still "Operating system" it should avoid to run Translator::instance()->setLocale(). This is because after all the issues we had with InputField we cannot be 100% sure that calling it causes some unknown side-effect.

0penBrain replied:
OK, I see. Will have a look now.

0penBrain replied:
@wwmayer Should be fixed now

freecadci commented:
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_6669"><img alt="pipeline status" src="https://gitlab.com/freecad/FreeCAD-CI/badges/PR_6669/pipeline.svg"/></a> for feature branch [PR_6669](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_6669). Pipeline [503438464](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/503438464) was triggered at [b748e84](https://github.com/FreeCAD/FreeCAD/pull/6669/commits/b748e84e9a795071564829d057b0b609433f8343). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

wwmayer commented:
> As a sidenote, what would you think if [KeyboardFilter::eventFilter()](https://github.com/FreeCAD/FreeCAD/blob/0b97a54e83c75da371b8f388468ae6d5c72451b5/src/Gui/GuiApplication.h#L101) would be moved as a member of Translator class (handling decimal separator conversion as of today)

It doesn't really fit to have it a part of that class because it's not directly related to languages. But I don't mind if you moved the code from Application.cpp to Translator.cpp
